test/regression.js

```diff
@@ -124,54 +120,52 @@ const readdirRecursive = async (absolute, relative = '') => {
   return result;
 };
 
+const width = 960;
+const height = 720;
 (async () => {
   try {
     const start = process.hrtime.bigint();
     const fixturesDir = path.join(__dirname, 'regression-fixtures');
     const list = await readdirRecursive(fixturesDir);
-    const originalFiles = new Map();
-    const optimizedFiles = new Map();
-    // read original and optimize
-    let failed = 0;
-    for (const name of list) {
-      try {
-        const file = path.join(fixturesDir, name);
-        const original = await fs.promises.readFile(file, 'utf-8');
-        const result = optimize(original, { path: name, floatPrecision: 4 });
-        if (result.error) {
-          console.error(result.error);
-          console.error(`File: ${name}`);
-          failed += 1;
-        } else {
-          originalFiles.set(name, original);
-          optimizedFiles.set(name, result.data);
-        }
-      } catch (error) {
-        console.error(error);
-        console.error(`File: ${name}`);
-        failed += 1;
-      }
-    }
-    if (failed !== 0) {
-      throw Error(`Failed to optimize ${failed} cases`);
-    }
     // setup server
-    const server = http.createServer((req, res) => {
-      if (req.url.startsWith('/original/')) {
-        const name = req.url.slice('/original/'.length);
-        if (originalFiles.has(name)) {
-          res.setHeader('Content-Type', 'image/svg+xml');
-          res.end(originalFiles.get(name));
+    const server = http.createServer(async (req, res) => {
+      const name = req.url.startsWith('/original/')
+        ? req.url.slice('/original/'.length)
+        : req.url.startsWith('/optimized/')
+          ? req.url.slice('/optimized/'.length)
+          : undefined;
```

Comment:
```suggestion
      const url = req.url;
      const name = url.slice(url.indexOf("/", 1));
```

This works for both cases, and we don't have to worry about `undefined` since this is only used internally.

Reply:
The logic you replaced it with isn't the same as my suggestion!

The filename can have path separators in it, so it's important to grab everything after the second separator, not just split.

The reason the tests pass now is because of the logic later that returns a 404 with empty content if we fail to read the file. (Which is bizarre in my opinion, so let's get rid of that.)

```suggestion
      const name = req.url.split('/')[2];
```